lab_3/main.py

```diff
@@ -20,7 +20,30 @@ def tokenize_by_sentence(text: str) -> tuple:
          (('_', 'h', 'e', '_'), ('_', 'i', 's', '_'), ('_', 'h', 'a', 'p', 'p', 'y', '_'))
          )
     """
-    pass
+    t = text.split()
+    output_text = []
+    start_sentence = 0
+    for i, word in enumerate(t):
+        sentence_list = []
+        if word[-1] in "!?." and (t[-1] == word or t[i + 1][0].isupper() is True or t[i + 1][0].isdigit() is True):
```

Comment:
Better to define the constant strings before the loop - at the start of the function.